service/gov_pay.go

```diff
@@ -141,6 +141,44 @@ func (gp *GovPayService) getGovPayPaymentState(paymentResource *models.PaymentRe
 	return &govPayResponse.State, Success, nil
 }
 
+// GetGovPayPaymentDetails gets the details of a GovPay payment
+func (gp *GovPayService) GetGovPayPaymentDetails(paymentResource *models.PaymentResourceRest) (*models.PaymentDetails, ResponseType, error) {
+	request, err := http.NewRequest("GET", paymentResource.MetaData.ExternalPaymentStatusURI, nil)
```

Comment:
we should probably check whether `paymentResource.MetaData.ExternalPaymentStatusURI` is defined (or is equals to empty string) before creating a request from it - the godoc for NewRequest doesn't explicitly say what it'll do if the url is an empty string

Reply:
`NewRequest` will return an error `empty url` if the URL is nil:
https://golang.org/src/net/url/url.go?s=14561:14596